services/kvstore/src/main/java/org/hyperledger/besu/services/kvstore/LayeredKeyValueStorage.java

```diff
@@ -116,28 +124,67 @@ public Optional<NearestKeyValue> getNearestTo(
   @Override
   public Stream<Pair<byte[], byte[]>> stream(final SegmentIdentifier segmentId) {
     throwIfClosed();
+    var ourLayerState = hashValueStore.computeIfAbsent(segmentId, s -> newSegmentMap());
 
```

Comment:
why not keeping the 
```java
final Lock lock = rwLock.readLock();
    lock.lock();
```
in this method ?

Reply:
This might be a rebase snafu - I am looking closer

Reply:
Looking back, I was probably thinking about Layered only in the context of frozen states.  I believe that is the case anywhere that we are streaming - snap server and snap client.  AFAIK only snap server would stream a layered worldstate.  

But you are right, it is feasible we could have a concurrent modification while we are streaming and perhaps some future code might not expect this.  Perhaps snap server and snap client could use an unsafeStream() method that does not do locking, and we can make stream() do rw locking.  

I suspect that rw lock will be fairly expensive to the performance of snap server, since we want to stream requests concurrently.  AFAIK the only code path that could potentially stream a layer is snap server.  I am open to suggestions here.

Reply:
If there is no choice for performance. I think UnsafeStream should be good